test/core/unit/engine_http.test.js

```diff
@@ -247,21 +435,152 @@ test('hooks - afterResponse', (t) => {
       t.fail();
     }
 
-    t.assert(finalContext.answer === answer);
+    t.ok(finalContext.answer === answer);
 
     t.end();
   });
 });
 
-test('Redirects', (t) => {
-  const target = nock('http://localhost:8888')
-    .get('/foo')
-    .reply(302, undefined, {
-      Location: '/bar'
-    })
-    .get('/bar')
-    .reply(200, { foo: 'bar' });
+test('hooks - beforeScenario', (t) => {
+  const endpoint = '/products';
+  const script = {
+    config: {
+      target: 'http://localhost:8888',
+      processor: {
+        setEndpoint: function (context, ee, next) {
+          context.vars.endpoint = endpoint;
+
+          t.ok(
+            ee instanceof EventEmitter,
+            'processor function should receive an event emitter'
+          );
+
+          return next();
+        }
+      }
+    },
+    scenarios: [
+      {
+        beforeScenario: 'setEndpoint',
+        flow: [
+          {
+            get: {
+              uri: '{{ endpoint }}'
+            }
+          }
+        ]
+      }
+    ]
+  };
+
+  const target = nock(script.config.target).get(endpoint).reply(200, 'ok');
+
+  const engine = new HttpEngine(script);
+  const ee = new EventEmitter();
+  const runScenario = engine.createScenario(script.scenarios[0], ee);
+
+  const initialContext = {
+    vars: {}
+  };
+
+  runScenario(initialContext, function userDone(err, finalContext) {
+    if (err) {
+      t.fail();
+    }
+
+    t.equal(
+      finalContext.vars.endpoint,
+      endpoint,
+      'it should set context vars before running the scenario'
+    );
+
+    t.ok(target.isDone(), `Should have made a request to ${endpoint}`);
+
+    t.end();
+  });
+});
+
+test('hooks - afterScenario', (t) => {
+  const endpoint = '/products';
+  const productsCount = 123;
+  const script = {
+    config: {
+      target: 'http://localhost:8888',
+      processor: {
+        checkProductsCount: function (context, ee, next) {
+          t.ok(
```

Comment:
as above - if the reason for this check is to check the interface of hook functions then we should check other args + maybe do it in one place to DRY?